DigitalLearningSolutions.Web.Tests/Helpers/EditCourseSectionHelperTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Helpers
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Web.Controllers.TrackingSystem.CourseSetup;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseContent;
+    using FluentAssertions;
+    using FluentAssertions.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    public class EditCourseSectionHelperTests
+    {
+        [Test]
+        public void
+            ProcessBulkSelect_with_select_all_learning_action_selects_all_learning_and_returns_null()
+        {
+            // Given
+            var formData = GetDefaultEditCourseSectionFormData();
+
+            // When
+            var result = EditCourseSectionHelper.ProcessBulkSelect(
+                formData,
+                EditCourseSectionHelper.SelectAllLearningAction
+            );
+
+            // Then
+            result.Should().BeNull();
+            formData.Tutorials
+                .All(t => t.LearningEnabled)
+                .Should().BeTrue();
+        }
+
+        [Test]
+        public void
+            ProcessBulkSelect_with_select_all_diagnostic_action_selects_all_diagnostic_and_returns_null()
+        {
+            // Given
+            var formData = GetDefaultEditCourseSectionFormData();
+
+            // When
+            var result = EditCourseSectionHelper.ProcessBulkSelect(
+                formData,
+                EditCourseSectionHelper.SelectAllDiagnosticAction
+            );
+
+            // Then
+            result.Should().BeNull();
+            formData.Tutorials
+                .All(t => t.DiagnosticEnabled)
+                .Should().BeTrue();
+        }
+
+        [Test]
+        public void
+            ProcessBulkSelect_with_deselect_all_learning_action_deselects_all_learning_and_returns_null()
+        {
+            // Given
+            var formData = GetDefaultEditCourseSectionFormData();
+
+            // When
+            var result = EditCourseSectionHelper.ProcessBulkSelect(
+                formData,
+                EditCourseSectionHelper.DeselectAllLearningAction
+            );
+
+            // Then
+            result.Should().BeNull();
+            formData.Tutorials
+                .All(t => !t.LearningEnabled)
+                .Should().BeTrue();
+        }
+
+        [Test]
+        public void
+            ProcessBulkSelect_with_deselect_all_diagnostic_action_deselects_all_diagnostic_and_returns_null()
+        {
+            // Given
+            var formData = GetDefaultEditCourseSectionFormData();
+
+            // When
+            var result = EditCourseSectionHelper.ProcessBulkSelect(
+                formData,
+                EditCourseSectionHelper.DeselectAllDiagnosticAction
+            );
+
+            // Then
+            result.Should().BeNull();
+            formData.Tutorials

```

Comment:
This can be written as `formData.Tutorials.Should().OnlyContain(t => !t.DiagnosticEnabled);`
OR
`formData.Tutorials.Should().OnlyContain(t => t.DiagnosticEnabled== false);` for easier reading on the negation.

Reply:
changed